packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.base.tsx

```diff
@@ -0,0 +1,206 @@
+import * as React from 'react';
+import { ISearchBoxProps, ISearchBoxStyleProps, ISearchBoxStyles } from './SearchBox.types';
+import {
+  BaseComponent,
+  autobind,
+  css,
+  getId,
+  KeyCodes,
+  customizable,
+  classNamesFunction
+} from '../../Utilities';
+
+import { IconButton } from '../../Button';
+import { Icon } from '../../Icon';
+
+import { FocusZone } from '../../FocusZone';
+
+const getClassNames = classNamesFunction<ISearchBoxStyleProps, ISearchBoxStyles>();
+
+
+export interface ISearchBoxState {
+  value?: string;
+  hasFocus?: boolean;
+  id?: string;
+}
+
+@customizable('SearchBox', ['theme'])
+export class SearchBoxBase extends BaseComponent<ISearchBoxProps, ISearchBoxState> {
+  public static defaultProps: ISearchBoxProps = {
+    labelText: 'Search',
+  };
+
+  private _rootElement: HTMLElement;
+  private _inputElement: HTMLInputElement;
+  private _latestValue: string;
+
+  public constructor(props: ISearchBoxProps) {
+    super(props);
+
+    this.state = {
+      value: props.value || '',
+      hasFocus: false,
+      id: getId('SearchBox')
+    };
+  }
+
+  public componentWillReceiveProps(newProps: ISearchBoxProps) {
+    if (newProps.value !== undefined) {
+      this._latestValue = newProps.value;
+      this.setState({
+        value: newProps.value
+      });
+    }
+  }
+
+  public render() {
+    let { labelText, className, disabled, underlined, getStyles, theme } = this.props;
+    let { value, hasFocus, id } = this.state;
+
+    const classNames = getClassNames(getStyles!, { theme: theme!, underlined, hasFocus, disabled });
+
+    return (
+      <div
+        ref={ this._resolveRef('_rootElement') }
+        className={ css(className, classNames.root,
+          hasFocus && 'is-active',
+          disabled && 'is-disabled',
+          value!.length > 0 && 'can-clear',
+          underlined && 'is-underlined'
+        ) }
+        { ...{ onFocusCapture: this._onFocusCapture } }
+      >
+        <div className={ css(classNames.iconContainer) }>
+          <Icon className={ css(classNames.icon) } iconName='Search' />
+        </div>
+        <input
+          id={ id }
+          className={ css(classNames.field) }
+          placeholder={ labelText }
+          onChange={ this._onInputChange }
+          onInput={ this._onInputChange }
+          onKeyDown={ this._onKeyDown }
+          value={ value }
+          disabled={ this.props.disabled }
+          aria-label={ this.props.ariaLabel ? this.props.ariaLabel : this.props.labelText }
+          ref={ this._resolveRef('_inputElement') }
+        />
+        { value!.length > 0 &&
+          <div className={ css(classNames.clearButton) }>
+            <IconButton styles={ { root: { height: 'auto' }, icon: { fontSize: '12px' } } } onClick={ this._onClearClick } iconProps={ { iconName: 'Clear' } } />
```

Comment:
Should the styles here also come from the styles file as a classname?

Reply:
As a button this really is just a placeholder for once we have the new button getStyles API. My guess is that this'll be some sort of nested getStyles.